pkg/service/roomservice.go

```diff
@@ -219,3 +276,24 @@ func (s *RoomService) writeMessage(ctx context.Context, room, identity string, m
 	msg.ParticipantKey = routing.ParticipantKey(room, identity)
 	return rtcSink.WriteMessage(msg)
 }
+
+func (s *RoomService) reserveRecorder(ctx context.Context, req *livekit.RecordRoomRequest) (string, error) {
+	id := utils.NewGuid(utils.RecordingPrefix)
+	reservation := &livekit.RecordingReservation{
+		Id:          id,
+		SubmittedAt: time.Now().UnixNano(),
+		Input:       req.Input,
+		Output:      req.Output,
+	}
+	b, err := proto.Marshal(reservation)
+	if err != nil {
+		return "", err
+	}
+
+	err = s.recorder.ReserveRecorder(ctx, string(b), id)
```

Comment:
seems cleaner to have Recorder to be responsible for serialization. interfaces that use structs seem easier to understand/test.

Reply:
I was getting some error about the proto already being initialized when importing the recorder package from the worker, but it seems ok now. Might've just been out of sync